File: src-tauri/src/commands/shortcut.rs
// Copyright (c) 2026. 千诚. Licensed under GPL v3.

//! 快捷键命令。
use tauri::AppHandle;

#[tauri::command]
pub fn register_global_shortcut(app: AppHandle, shortcut: String) -> Result<(), String> {
    crate::core::system::shortcut::register_global_shortcut(app, shortcut)
}

#[tauri::command]
pub fn get_shortcut_status() -> (bool, Option<String>) {
    crate::core::system::shortcut::get_shortcut_status()
}

File: src/services/NativeService/shortcut.ts
import { invoke } from '@tauri-apps/api/core';

export const shortcut = {
    registerGlobalShortcut(shortcut: string): Promise<void> {
        return invoke('register_global_shortcut', { shortcut });
    },
    getShortcutStatus(): Promise<[boolean, string | null]> {
        return invoke('get_shortcut_status');
    },
} as const;
